### ISSUE TITLE
Figure subcaptions are randomly arranged in an order I didn't set in a python chunk in a qmd file.

### ISSUE THREAD

**Issue by kmink3225** [quarto-dev/quarto-cli#5497]
### Bug description

````
---
title-block-banner: true
title: "Something1"
subtitle: "Something2"
date: 2023-04-03
author: "Someone"
institute: "Something3"
format:
  html:
    toc: true
    number-sections: true
    lot: true
    lof: true
    code-fold: true
    fig-width: 15
    fig-height: 15
execute:
  warning: false
  message: false
  echo: false
---
````
````

```{python, error_signals}
#| label: error_signals
#| fig-cap: "Error Signals Grouped By Targets"
#| fig-subcap: 
#|   - "a"
#|   - "b"
#|   - "c"
#|   - "d"
#|   - "e"
#|   - "f"
#| layout-ncol: 2

# for target in targets:
#   temp_dataid=unnested_error_data[unnested_error_data.target==target].dataid.unique()
#   plot_error_signals(temp_dataid,title=target)
#   plt.show()

plt.plot([1,23,2,4])
plt.show()

plt.plot([8,65,23,90])
plt.show()

plt.plot([1,3,2,4])
plt.show()

plt.plot([1,1,1,1])
plt.show()

plt.plot([1,2,3,4])
plt.show()

plt.plot([4,3,2,1])
plt.show()
```

````

the rendered output:

![image](https://github.com/quarto-dev/quarto-cli/assets/61118160/6ba16dd3-0772-44d1-992c-61949a029421)

### Checklist

- [ ] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [ ] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [ ] Please document the operating system you're running. If on Linux, please provide the specific distribution.
- [ ] Please provide the output of `quarto check` so we know which version of quarto and its dependencies you're running.

**Comment by mcanouil** [comment]
Thanks for the report, unfortunately, your code is not reproducible.
Please follow the guidelines and provide a complete (not) working Quarto document for us to work with.
Also, please read the checklist and provide the requested informations.

**Comment by mcanouil** [comment]
I managed to get the missing part at least to run the code, but I can't reproduce the issue using Quarto 1.3.340 or current development version with the following simplified example (same results with your big yaml header).

<details><summary>See Quarto document code.</summary>

````qmd
---
format: html
---

```{python}
#| label: error_signals
#| fig-cap: "Error Signals Grouped By Targets"
#| fig-subcap: 
#|   - "a"
#|   - "b"
#|   - "c"
#|   - "d"
#|   - "e"
#|   - "f"
#| layout-ncol: 2

import matplotlib.pyplot as plt

plt.plot([1,23,2,4])
plt.show()

plt.plot([8,65,23,90])
plt.show()

plt.plot([1,3,2,4])
plt.show()

plt.plot([1,1,1,1])
plt.show()

plt.plot([1,2,3,4])
plt.show()

plt.plot([4,3,2,1])
plt.show()
```
````

</details>

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/373de468-a656-4127-be6e-2ad26dc4a3ea">

**Comment by kmink3225** [comment]
wow! Thanks for the really quick reply. I have a long code, and because of security issues, I have to select only part of it and give you a code to reproduce with some simulated data, but it's not easy because I'm not that good at Python. (Even, I generated some fake data and functions with the same structure that causes this problem, but I failed to reproduce the problem..). 

But because of your code, at least I was able to figure out which part was causing this problem. This is not an issue with quarto, it seems to be caused by another function call I made and the way I preprocessed the data previously. I guess it will take some time to figure out which line specifically is causing the problem. This is my first time submitting an issue, so I apologize for the inappropriate issue submission form. Next time, I will stick to the guidelines and checklist. Thanks for your answer.
